src/plugins/resolver/filename.c

```diff
@@ -5,23 +5,27 @@
  *
  * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
  */
+#define _POSIX_SOURCE
```

Comment:
Looks like you also need to define the minimum POSIX level you need on macOS:

```c
#define _POSIX_SOURCE
#define _POSIX_C_SOURCE 200409L 
```

.